tests/devlxd-container

```diff
@@ -0,0 +1,134 @@
+fi
+lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock http://custom.socket/1.0/config | jq
+lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock http://custom.socket/1.0/meta-data | grep -q '#cloud-config'
+
+# Disable devlxd
+lxc config set c1 security.devlxd false
+
+echo "==> Checking devlxd is not working"
+
+! lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock http://custom.socket/1.0 || false
+
+lxc restart -f c1
+
+# devlxd should not be running after restart
+! lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock http://custom.socket/1.0 || false
+
+echo "==> Checking devlxd can be enabled live"
+
+# Enable devlxd
+lxc config set c1 security.devlxd true
+if hasNeededAPIExtension devlxd_images; then
+  lxc config set c1 security.devlxd.images true
+fi
+
+# devlxd should be running after the config is enabled
+lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock http://custom.socket/1.0 | jq
+if hasNeededAPIExtension cloud_init; then
+  lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock http://custom.socket/1.0/devices | jq
+fi
+lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock http://custom.socket/1.0/config | jq
+lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock http://custom.socket/1.0/meta-data | grep -q '#cloud-config'
+
+if hasNeededAPIExtension instance_ready_state; then
+  # test instance Ready state
+  lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock -X PATCH -d '{"state":"Ready"}' http://custom.socket/1.0
+  [ "$(lxc config get c1 volatile.last_state.ready)" = "true" ]
+
+  lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock -X PATCH -d '{"state":"Started"}' http://custom.socket/1.0
+  [ "$(lxc config get c1 volatile.last_state.ready)" = "false" ]
+
+  lxc exec c1 -- curl -s --unix-socket /dev/lxd/sock -X PATCH -d '{"state":"Ready"}' http://custom.socket/1.0
+  [ "$(lxc config get c1 volatile.last_state.ready)" = "true" ]
+  lxc stop -f c1
+  sleep 5
+  [ "$(lxc config get c1 volatile.last_state.ready)" = "false" ]
+
+  lxc start c1
+else
+  echo "Skipping instance Ready state tests, not supported"
+fi
+
+# Below tests for image export rely on an event being emitted on export. This will not be backported to 4.0 or 5.0,
+# so exit the test here.
+if [ "${LXD_SNAP_CHANNEL}" = "4.0/edge" ] || [ "${LXD_SNAP_CHANNEL}" = "5.0/edge" ]; then
+  echo "Skipping devlxd image export tests, not supported"
+  FAIL=0
+  exit
+fi
+
+# We need snapd to be ready before the next tests. Finagle the waitSnapdSeed function definition into the container and run it.
+lxc exec c1 -- sh -c "$(type waitSnapdSeed | tail -n +2); waitSnapdSeed"
+lxc exec c1 -- snap remove --purge lxd || true
+lxc exec c1 -- snap install lxd --channel="${LXD_SNAP_CHANNEL}"
+lxc exec c1 -- /snap/bin/lxd init --auto
+
+if hasNeededAPIExtension devlxd_images && hasNeededAPIExtension event_lifecycle; then
+  lxc monitor --type lifecycle --format json > monitor.json 2>&1 &
+  echo $! > monitor.pid
```

Comment:
Can you just store this as a variable rather than a file, less to clear up or be left behind.